controllers/apps/operations/switchover.go

```diff
@@ -0,0 +1,307 @@
+			opsRes.OpsRequest.Status.Components[compSpecName] = appsv1alpha1.OpsRequestComponentStatus{
+				Phase:           appsv1alpha1.SpecReconcilingClusterCompPhase,
+				ProgressDetails: componentProcessDetails,
+			}
+		}
+
+		// check the current component roleLabel whether correct
+		checkRoleLabelProcessDetail := appsv1alpha1.ProgressStatusDetail{
+			ObjectKey: fmt.Sprintf("%s/%s", SwitchoverCheckRoleLabelKey, compSpecName),
+			Status:    appsv1alpha1.ProcessingProgressStatus,
+			Message:   fmt.Sprintf("waiting for component %s pod role label consistency after switchover", compSpecName),
+		}
+		compDef, err = componentutil.GetComponentDefByCluster(reqCtx.Ctx, cli, *opsRes.Cluster, compSpecName)
+		if err != nil {
+			checkRoleLabelProcessDetail.Message = fmt.Sprintf("handleSwitchoverProgress get component %s definition failed", compSpecName)
+			checkRoleLabelProcessDetail.Status = appsv1alpha1.FailedProgressStatus
+			setComponentStatusProgressDetail(reqCtx.Recorder, opsRequest, &componentProcessDetails, checkRoleLabelProcessDetail)
+			opsRes.OpsRequest.Status.Components[compSpecName] = appsv1alpha1.OpsRequestComponentStatus{
+				Phase:           appsv1alpha1.SpecReconcilingClusterCompPhase,
+				ProgressDetails: componentProcessDetails,
+			}
+			continue
+		}
+		consistency, err = checkPodRoleLabelConsistency(reqCtx.Ctx, cli, opsRes.Cluster, opsRes.Cluster.Spec.GetComponentByName(compSpecName), compDef, &switchover, switchoverCondition)
+		if err != nil {
+			checkRoleLabelProcessDetail.Message = fmt.Sprintf("waiting for component %s pod role label consistency after switchover", compSpecName)
+			setComponentStatusProgressDetail(reqCtx.Recorder, opsRequest, &componentProcessDetails, checkRoleLabelProcessDetail)
+			opsRes.OpsRequest.Status.Components[compSpecName] = appsv1alpha1.OpsRequestComponentStatus{
+				Phase:           appsv1alpha1.SpecReconcilingClusterCompPhase,
+				ProgressDetails: componentProcessDetails,
+			}
+			continue
+		}
+
+		if !consistency {
+			err = intctrlutil.NewErrorf(intctrlutil.ErrorWaitCacheRefresh, "requeue to waiting for pod role label consistency.")
+			opsRes.OpsRequest.Status.Components[compSpecName] = appsv1alpha1.OpsRequestComponentStatus{
+				Phase:           appsv1alpha1.SpecReconcilingClusterCompPhase,
+				ProgressDetails: componentProcessDetails,
+			}
+			continue
+		} else {
+			checkRoleLabelProcessDetail.Message = fmt.Sprintf("check component %s pod role label consistency after switchover is succeed", compSpecName)
+			checkRoleLabelProcessDetail.Status = appsv1alpha1.SucceedProgressStatus
+			setComponentStatusProgressDetail(reqCtx.Recorder, opsRequest, &componentProcessDetails, checkRoleLabelProcessDetail)
+			opsRes.OpsRequest.Status.Components[compSpecName] = appsv1alpha1.OpsRequestComponentStatus{
+				Phase:           appsv1alpha1.SpecReconcilingClusterCompPhase,
+				ProgressDetails: componentProcessDetails,
+			}
```

Comment:
whether can abstract a unified  function to set status.components[i].ProgressDetail?